release/azure-pipelines-release.yml

```diff
@@ -98,6 +99,22 @@ jobs:
       cat $OBJ_DIR/hw/top_earlgrey/chip_earlgrey_cw310_hyperdebug-release/build.fpga_cw310/synth-vivado/lowrisc_systems_chip_earlgrey_cw310_0.1.runs/impl_1/runme.log || true
     displayName: "Display hyperdebug bitstream synth and impl logs"
 
+  - task: GithubRelease@1
+    displayName: "Publish release to Github"
+    inputs:
+      gitHubConnection: "opentitan-github-connection" # TODO(milesdai): create the service connection before merging this PR
+      repositoryName: "$(Build.Repository.Name)"
+      action: "create"
+      target: "$(Build.SourceVersion)"
+      tagSource: "userSpecifiedTag"
+      tag: "snapshot-$(RELEASE_DATE)"
+      releaseNotesSource: "inline"
+      releaseNotesInline: "Nightly release"
+      assets: "$(Build.SourcesDirectory)/$(pkg_path)"
```

Comment:
Can `assets` be a list?

Right now, there is only one release artifact (the output of `//release:opentitan`).  We may want multiple artifacts in the future.